### DISCUSSION TITLE
Need mutation example on how to use graphql-request sdk

### DISCUSSION THREAD

**Discussion by RedFour** [dotansimha/graphql-code-generator#8422] [Q&A] [answered]
The example provide here only has a query.

https://www.the-guild.dev/graphql/codegen/plugins/typescript/typescript-graphql-request

Can someone provide me with an example showing how to use it for mutation with variables?

**Comment by charlypoly** [comment, marked as answer]
Hi @RedFour,

A mutation has the same signature as a query, with variables as the first argument of the SDK method, as shown below:

```ts
import { GraphQLClient } from 'graphql-request'
import { getSdk } from './sdk' // THIS FILE IS THE GENERATED FILE
 
async function main() {
  const client = new GraphQLClient('https://countries.trevorblades.com')
  const sdk = getSdk(client)
  const { continent } = await sdk.createContinent({ name: "Europe" }) // This is fully typed, based on the query
 
  console.log(`GraphQL data:`, continent)
}
```
